framework/test/unit/modules/interoperability/mainchain/commands/terminate_sidechain_for_liveness.spec.ts

```diff
@@ -0,0 +1,197 @@
+			expect(result.error?.message).toInclude('Chain account does not exist');
+		});
+
+		it('should return error when chain is already terminated', async () => {
+			await interopMod.stores
+				.get(ChainAccountStore)
+				.set(commandVerifyContext as any, transactionParams.chainID, {
+					...chainAccount,
+					status: ChainStatus.TERMINATED,
+				});
+
+			const result = await livenessTerminationCommand.verify(commandVerifyContext);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error?.message).toInclude('Sidechain is already terminated');
+		});
+
+		it('should return error if the chain is not live', async () => {
+			jest.spyOn(interopMod['internalMethod'], 'isLive').mockResolvedValue(false);
+
+			const result = await livenessTerminationCommand.verify(commandVerifyContext);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error?.message).toInclude('Sidechain did not violate the liveness condition');
+		});
+
+		it('should return VerifyStatus.OK when chain is active', async () => {
+			const result = await livenessTerminationCommand.verify(commandVerifyContext);
+
+			expect(result.status).toBe(VerifyStatus.OK);
+		});
+	});
+
+	describe('execute', () => {
+		let livenessTerminationCommand: TerminateSidechainForLivenessCommand;
+		let commandExecuteContext: CommandExecuteContext<TerminateSidehchainForLivenessParams>;
+		let interoperableCCMethods: Map<string, BaseCCMethod>;
+		let ccCommands: Map<string, BaseCCCommand[]>;
+		let transaction: Transaction;
+		let transactionParams: TerminateSidehchainForLivenessParams;
+		let encodedTransactionParams: Buffer;
+		let transactionContext: TransactionContext;
+
+		beforeEach(() => {
+			interoperableCCMethods = new Map();
+			ccCommands = new Map();
+			livenessTerminationCommand = new TerminateSidechainForLivenessCommand(
+				interopMod.stores,
+				interopMod.events,
+				interoperableCCMethods,
+				ccCommands,
+				interopMod['internalMethod'],
+			);
+
+			transactionParams = {
+				chainID: utils.intToBuffer(3, 4),
+			};
+
+			encodedTransactionParams = codec.encode(
+				terminateSidechainForLivenessParamsSchema,
+				transactionParams,
+			);
+
+			transaction = new Transaction({
+				module: MODULE_NAME_INTEROPERABILITY,
+				command: COMMAND_NAME_LIVENESS_TERMINATION,
+				fee: BigInt(100000000),
+				nonce: BigInt(0),
+				params: encodedTransactionParams,
+				senderPublicKey: utils.getRandomBytes(32),
+				signatures: [],
+			});
+
+			transactionContext = createTransactionContext({
+				transaction,
+			});
+
+			commandExecuteContext = transactionContext.createCommandExecuteContext<TerminateSidehchainForLivenessParams>(
+				terminateSidechainForLivenessParamsSchema,
+			);
+			jest.spyOn(interopMod['internalMethod'], 'terminateChainInternal').mockResolvedValue();
+		});
+
+		it('should successfully terminate chain', async () => {
+			await livenessTerminationCommand.execute(commandExecuteContext);
```

Comment:
it's good to expect the `terminateChainInternal` is called with expected chainID?